pkg/server/proto/server.proto

```diff
@@ -5094,6 +5122,89 @@ message ConfigSyncPipelineResponse {
   map<string, Ref.Pipeline> synced_pipelines = 1;
 }
 
+/********************************************************************
+* ProjectTemplate
+********************************************************************/
+
+message GetProjectTemplateRequest {
+  Ref.ProjectTemplate project_template = 1;
+}
+
+message GetProjectTemplateResponse {
+  // A single projecttemplate
+  ProjectTemplate project_template = 1;
+}
+
+message ListProjectTemplatesRequest {
+  PaginationRequest pagination = 1;
+}
+
+message ListProjectTemplatesResponse {
+  // The full list of projecttemplates
+  repeated ProjectTemplate project_templates = 1;
+  PaginationResponse pagination = 2;
+  uint64 total_count = 3;
+}
+
+message UpsertProjectTemplateRequest {
+  ProjectTemplate project_template = 1;
+}
+
+message UpsertProjectTemplateResponse {
+  ProjectTemplate project_template = 1;
+}
+
+message DeleteProjectTemplateRequest {
+  Ref.ProjectTemplate project_template = 1;
+}
+
+message DeleteProjectTemplateResponse {
+}
+
+message ProjectTemplate {
+  // Unique ID of the ProjectTemplate
+  string id = 1;
+
+  // Unique name of the ProjectTemplate
+  string name = 2;
+
+  // A brief summary for use by platform engineers to distinguish between
+  // multiple templates
+  string summary = 3;
+
+  // A long summary of what the ProjectTemplate is to be used for. This summary
+  // is shared between platform engineers and application developers.
+  string expanded_summary = 4;
+
+  // The template which is rendered when a Project is created from a
+  // ProjectTemplate to be shown to application developers.
+  string readme_template = 5;
+
+  // Settings for the Waypoint project that should be set when a project is
+  // created from a ProjectTemplate.
+  WaypointProject waypoint_project = 6;
+
+  // The Terraform No-Code Module which should be used to provision
+  // infrastructure that is used by the project created from a ProjectTemplate.
+  TerraformNocodeModule terraform_nocode_module = 7;
+
+  // A list of descriptive strings that can be applied to the ProjectTemplate.
+  repeated string tags = 8;
+
+  message WaypointProject {
+    // The templated waypoint.hcl file stored as HCL.
+    bytes waypoint_hcl_template = 1;
```

Comment:
I'm a bit confused on the naming here, the upper level item is a `ProjectTemplate`, which contains a type `WaypointProject` which I would expect by the name to reference a project in Waypoint, but instead it's a HCL template file?

Reply:
Maybe we should add a description to the message that says something like:

```
WaypointProject governs the properties that will be set on the final waypoint Project that we create. Future fields will likely include status report and datasource polling settings.
```